CTFd/api/v1/helpers/request.py

```diff
@@ -96,3 +119,23 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     return decorator
+
+
+def validate_args(spec, location, allow_extras=False):
+    """
+    Validate and load arguments according to a pydantic spec.
+
+    :param spec: A pydantic model or a dictionary of fields.
+    :param location: A string indicating where to find the arguments. Can be one of "json", "query", "form", "headers", "cookies"
+    :param allow_extras: Whether to allow extra arguments not defined in the spec.
+    :return: A decorator that validates and loads arguments according to the spec.
+
+    Example usage:
+
+    .. code-block:: python
+
+        @validate_args({"name": (str, None), "id": (int, None)}, location="query")
+        def my_route(query_args):
+            return {"success": True, "name": query_args["name"], "id": query_args["id"]}
+    """
+    return expects_args(spec, location, allow_extras, validate=True)
```

Comment:
Shouldn't we put a "deprecated" something here ? To make sure plugin developers have the info of a breaking change in their IDE.
Or do we expect developers to not use the `validate_args` decorator ?

Reply:
Currently the decorator still has the same behavior as by default, just now with a new option to allow extras. We could mark it as depreciated, but I'll leave it up to @ColdHeat.